src/hydra/HydraAdmin.js

```diff
@@ -7,34 +7,70 @@ import restClient from './hydraClient';
 class HydraAdmin extends Component {
   static defaultProps = {
     apiDocumentationParser,
+    customRoutes: [],
+    error: 'Unable to retrieve API documentation.',
+    loading: 'Loading...',
     restClient,
   };
 
   static propTypes = {
     apiDocumentationParser: PropTypes.func,
+    customRoutes: PropTypes.array,
     entrypoint: PropTypes.string.isRequired,
+    error: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
+    loading: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
     restClient: PropTypes.func,
   };
 
   state = {
     api: null,
+    customRoutes: [],
+    hasError: false,
+    loaded: false,
   };
 
   componentDidMount() {
     this.props
       .apiDocumentationParser(this.props.entrypoint)
-      .then(api => this.setState({api}));
+      .then(
+        ({api, customRoutes = []}) => ({
+          api,
+          customRoutes,
+          hasError: false,
+          loaded: true,
+        }),
+        ({api, customRoutes = []}) => ({
+          api,
+          customRoutes,
+          hasError: true,
+          loaded: true,
+        }),
+      )
+      .then(state => this.setState(state));
```

Comment:
Have you tried `.then(this.setState)` ?

Reply:
I tried and the following error occurred:

```
Uncaught (in promise) TypeError: Cannot read property 'updater' of undefined
```

Reply:
It works, if I use:

```js
.then(this.setState.bind(this))
```